lib/service_check.pm

```diff
@@ -139,16 +163,45 @@ sub install_services {
 }
 
 sub check_services {
-    my $service = shift;
+    my ($service) = @_;
+    my $service_func;
     foreach my $s (keys %$service) {
         my $srv_proc_name = $service->{$s}->{srv_proc_name};
         my $support_ver   = $service->{$s}->{support_ver};
         if (grep { $_ eq $hdd_base_version } split(',', $support_ver)) {
+            # service check after migration. if the we've set up service check
+            # function, we don't need following actions to check the service.
+            if (exists $service->{$s}->{service_check_func}) {
+                $service_func = $service->{$s}->{service_check_func};
+                call_func(\&$service_func, 'after');
+                next;
+            }
+
             assert_script_run 'systemctl status '
               . $srv_proc_name
-              . ' --no-pager | grep active';
+              . ' --no-pager | grep -w active';
         }
     }
 }
 
+#
+# check apache service before and after migration
+#
+sub full_apache_check {
+    my ($stage) = @_;
+    if ($stage eq 'before') {
+        # check service before migration
+        http_srv::check_apache_install();
```

Comment:
I think we should rather have this methods in the library instead of importing them from the test modules

Reply:
in the prototype we discussed is that we use the existed modules to re-construct what we needs
in the service check function.